File: expedite-consults/mail-platform/frontend/src/types/mail.ts
export type MailboxRole = 'inbox' | 'starred' | 'sent' | 'drafts' | 'spam' | 'trash' | 'archive' | 'custom';
export type AICategory = 'primary' | 'updates' | 'promotions' | 'urgent';

export interface Attachment {
  id: string;
  filename: string;
  content_type: string;
  size_bytes: number;
  s3_key: string;
  is_inline: boolean;
}

export interface Message {
  id: string;
  thread_id: string;
  message_id: string;
  in_reply_to?: string;
  from_address: string;
  from_name?: string;
  to_addresses: string[];
  cc_addresses?: string[];
  bcc_addresses?: string[];
  subject: string;
  body_plain?: string;
  body_html?: string;
  snippet?: string;
  is_read: boolean;
  is_starred: boolean;
  is_draft: boolean;
  has_attachments: boolean;
  ai_summary?: string;
  sent_at?: string;
  received_at: string;
  attachments?: Attachment[];
}

export interface Thread {
  id: string;
  subject: string;
  snippet?: string;
  last_message_at: string;
  message_count: number;
  is_read: boolean;
  is_starred: boolean;
  has_attachments: boolean;
  ai_summary?: string;
  ai_category: AICategory;
  created_at: string;
}

export interface ThreadDetail extends Thread {
  messages: Message[];
}

export interface Mailbox {
  id: string;
  name: string;
  role: MailboxRole;
  color?: string;
  total_messages: number;
  unread_messages: number;
}

File: expedite-consults/mail-platform/backend/app/schemas/mail.py
from pydantic import BaseModel, EmailStr
from typing import List, Optional, Any
from uuid import UUID
from datetime import datetime

class AttachmentOut(BaseModel):
    id: UUID
    filename: str
    content_type: str
    size_bytes: int
    s3_key: str
    is_inline: bool

    class Config:
        from_attributes = True

class MessageOut(BaseModel):
    id: UUID
    thread_id: UUID
    message_id: str
    in_reply_to: Optional[str] = None
    from_address: str
    from_name: Optional[str] = None
    to_addresses: List[Any]
    cc_addresses: List[Any]
    bcc_addresses: List[Any]
    subject: str
    body_plain: Optional[str] = None
    body_html: Optional[str] = None
    snippet: Optional[str] = None
    is_read: bool
    is_starred: bool
    is_draft: bool
    has_attachments: bool
    ai_summary: Optional[str] = None
    sent_at: Optional[datetime] = None
    received_at: datetime
    attachments: List[AttachmentOut] = []

    class Config:
        from_attributes = True

class ThreadOut(BaseModel):
    id: UUID
    subject: str
    snippet: Optional[str] = None
    last_message_at: datetime
    message_count: int
    is_read: bool
    is_starred: bool
    has_attachments: bool
    ai_summary: Optional[str] = None
    ai_category: str
    created_at: datetime

    class Config:
        from_attributes = True

class ThreadDetailOut(ThreadOut):
    messages: List[MessageOut] = []

class MailboxOut(BaseModel):
    id: UUID
    name: str
    role: str
    color: Optional[str] = "#6B7280"
    total_messages: int
    unread_messages: int

    class Config:
        from_attributes = True

class SendMessageRequest(BaseModel):
    to_addresses: List[str]
    cc_addresses: Optional[List[str]] = []
    bcc_addresses: Optional[List[str]] = []
    subject: str
    body_html: str
    body_plain: Optional[str] = None
    in_reply_to: Optional[str] = None
    thread_id: Optional[UUID] = None
    attachment_ids: Optional[List[UUID]] = []

class SaveDraftRequest(BaseModel):
    draft_id: Optional[UUID] = None
    thread_id: Optional[UUID] = None
    to_addresses: Optional[List[str]] = []
    cc_addresses: Optional[List[str]] = []
    bcc_addresses: Optional[List[str]] = []
    subject: Optional[str] = ""
    body_html: Optional[str] = ""
    body_plain: Optional[str] = ""
